heapq failures on win32

## Reporting a bug

Observed on e050fd691abcde71ac84dd8f748d158332ff060c in buildfarm run. Buildfarm has not been auto testing mainline since 0.54 rc, so first failing commit could be earlier.

The errors
```pytb
======================================================================
ERROR: test_heapsort (numba.tests.test_heapq.TestHeapqReflectedList)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\test_heapq.py", line 397, in test_heapsort
    cfunc_heappush(heap, item)
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\core\dispatcher.py", line 482, in _compile_for_args
    error_rewrite(e, 'typing')
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\core\dispatcher.py", line 423, in error_rewrite
    raise e.with_traceback(None)
numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<built-in function heappush>) found for signature:

 >>> heappush(reflected list(int32)<iv=None>, int64)

There are 2 candidate implementations:
  - Of which 2 did not match due to:
  Overload in function 'heappush': File: numba\cpython\heapq.py: Line 150.
    With argument(s): '(reflected list(int32)<iv=None>, int64)':
   Rejected as the implementation raised a specific error:
     TypingError: heap type must be the same as item type
  raised from C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\cpython\heapq.py:119

During: resolving callee type: Function(<built-in function heappush>)
During: typing of call at C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\test_heapq.py (24)


File "..\..\..\..\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\test_heapq.py", line 24:
def heappush(heap, item):
    return hq.heappush(heap, item)
    ^


======================================================================
ERROR: test_nbest_with_pushpop (numba.tests.test_heapq.TestHeapqTypedList)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\test_heapq.py", line 445, in test_nbest_with_pushpop
    cfunc_heappushpop(heap, item)
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\core\dispatcher.py", line 482, in _compile_for_args
    error_rewrite(e, 'typing')
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\core\dispatcher.py", line 423, in error_rewrite
    raise e.with_traceback(None)
numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<built-in function heappushpop>) found for signature:

 >>> heappushpop(ListType[int64], int32)

There are 2 candidate implementations:
      - Of which 2 did not match due to:
      Overload in function 'heappushpop': File: numba\cpython\heapq.py: Line 176.
        With argument(s): '(ListType[int64], int32)':
       Rejected as the implementation raised a specific error:
         TypingError: heap type must be the same as item type
  raised from C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\cpython\heapq.py:119

During: resolving callee type: Function(<built-in function heappushpop>)
During: typing of call at C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\test_heapq.py (28)


File "..\..\..\..\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\test_heapq.py", line 28:
def heappushpop(heap, item):
    return hq.heappushpop(heap, item)
    ^


======================================================================
ERROR: test_nbest_with_pushpop (numba.tests.test_heapq.TestHeapqReflectedList)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\test_heapq.py", line 445, in test_nbest_with_pushpop
    cfunc_heappushpop(heap, item)
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\core\dispatcher.py", line 482, in _compile_for_args
    error_rewrite(e, 'typing')
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\core\dispatcher.py", line 423, in error_rewrite
    raise e.with_traceback(None)
numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<built-in function heappushpop>) found for signature:

 >>> heappushpop(reflected list(int32)<iv=None>, int64)

There are 2 candidate implementations:
  - Of which 2 did not match due to:
  Overload in function 'heappushpop': File: numba\cpython\heapq.py: Line 176.
    With argument(s): '(reflected list(int32)<iv=None>, int64)':
   Rejected as the implementation raised a specific error:
     TypingError: heap type must be the same as item type
  raised from C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\cpython\heapq.py:119

During: resolving callee type: Function(<built-in function heappushpop>)
During: typing of call at C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\test_heapq.py (28)


File "..\..\..\..\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\test_heapq.py", line 28:
def heappushpop(heap, item):
    return hq.heappushpop(heap, item)
    ^


======================================================================
FAIL: test_nbest_with_pushpop (numba.tests.test_heapq.TestHeapqTypedList)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\support.py", line 779, in tearDown
    self.memory_leak_teardown()
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\support.py", line 753, in memory_leak_teardown
    self.assert_no_memory_leak()
  File "C:\Users\ci\32\miniconda3\envs\testenv_0ca9fc0b-ca29-4f20-8382-ad13e833b078\lib\site-packages\numba\tests\support.py", line 762, in assert_no_memory_leak
    self.assertEqual(total_alloc, total_free)
AssertionError: 3 != 1

----------------------------------------------------------------------
```

Update 1:

- problem seems to be agnostic to python versions. failed on py3.7 and py3.9. py3.8 not tested in the farm.

--- a/numba/tests/test_heapq.py
+++ b/numba/tests/test_heapq.py
@@ -5,7 +5,6 @@ import numpy as np
 
 from numba import jit, typed
 from numba.core.compiler import Flags
-from numba.core.config import IS_WIN32
 from numba.tests.support import TestCase, CompilationCache, MemoryLeakMixin
 
 no_pyobj_flags = Flags()

--- a/numba/tests/test_heapq.py
+++ b/numba/tests/test_heapq.py
@@ -379,14 +379,10 @@ class _TestHeapq(MemoryLeakMixin):
         cfunc_heappush = jit(nopython=True)(heappush)
         cfunc_heappop = jit(nopython=True)(heappop)
 
-        arange_kwargs = {}
-        if IS_WIN32:
-            # Ensure consistency of typing on Windows, where the default dtype
-            # of arange is int32 (vs. int64 on other 64-bit platforms)
-            arange_kwargs['dtype'] = np.int64
-
         for trial in range(100):
-            values = np.arange(5, **arange_kwargs)
+            # Ensure consistency of typing, use float64 as it's double
+            # everywhere
+            values = np.arange(5, dtype=np.float64)
             data = self.listimpl(self.rnd.choice(values, 10))
             if trial & 1:
                 heap = data[:]
@@ -430,13 +426,8 @@ class _TestHeapq(MemoryLeakMixin):
         pyfunc_heapify = heapify
         cfunc_heapify = jit(nopython=True)(pyfunc_heapify)
 
-        arange_kwargs = {}
-        if IS_WIN32:
-            # Ensure consistency of typing on Windows, where the default dtype
-            # of arange is int32 (vs. int64 on other 64-bit platforms)
-            arange_kwargs['dtype'] = np.int64
-
-        values = np.arange(2000, **arange_kwargs)
+        # Ensure consistency of typing, use float64 as it's double everywhere
+        values = np.arange(2000, dtype=np.float64)
         data = self.listimpl(self.rnd.choice(values, 1000))
         heap = data[:10]
         cfunc_heapify(heap)

